core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -70,6 +70,26 @@ public Table loadTable(TableIdentifier ident) {
     return tableCache.get(canonicalizeIdentifier(ident), catalog::loadTable);
   }
 
+  public void createNamespace(Namespace namespace, Map<String, String> meta) {
+    catalog.createNamespace(namespace, meta);
+  }
+
+  public List<Namespace> listNamespaces() {
+    return catalog.listNamespaces();
+  }
+
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    return catalog.listNamespaces(namespace);
+  }
+
+  public Map<String, String>  loadNamespaceMetadata(Namespace namespace) {
+    return catalog.loadNamespaceMetadata(namespace);
+  }
+
+  public boolean dropNamespace(Namespace namespace) {
+    return catalog.dropNamespace(namespace);
```

Comment:
Because this drops objects in the namespace, it should also invalidate them in the cache.